HannaPidchasiuk/HW10/company.py

```diff
@@ -0,0 +1,118 @@
+import re
+
+
+class Company:
+
+    def __init__(self, name: str, number_of_employees: int, products: list[str], website: str):
+        self.__name = name
+        self.__number_of_employees = number_of_employees
+        self.__products = products
+        self.__website = website
+
+    @property
+    def name(self):
+        """
+            Returns name of the company.
+        """
+        return self.__name
+
+    @name.setter
+    def name(self, new_value: str):
+        """
+            Set new name of the worker.
+            Takes only 1 argument: new_value.
+        """
+        self.__name = new_value
+
+    @property
+    def number_of_employees(self):
+        """
+            Returns number of employees of the company.
+        """
+        return self.__number_of_employees
+
+    @number_of_employees.setter
+    def number_of_employees(self, new_value: int):
+        """
+            Set new number_of_employees of the company.
+            Takes only 1 argument: new_value.
+        """
+        if new_value < 0:
+            raise ValueError("Number of employees can't be less than 0!")
+        else:
+            self.__number_of_employees = new_value
+
+    @property
+    def products(self) -> list[str]:
+        """
+            Returns list of products of the company.
+        """
+        return self.__products
+
+    @products.setter
+    def products(self, new_value: str):
+        """
+            Add new product to products.
+            Takes only 1 argument: new_value.
+        """
+        self.__products.append(new_value)
+
+    @property
+    def website(self):
+        """
+            Returns web_site of the company.
+        """
+        return self.__website
+
+    @website.setter
+    def website(self, new_value: str):
+        """
+            Set new website of the company.
+            Takes only 1 argument: new_value.
+        """
+        self.__website = new_value
+
+    def add_many_products(self, *args):
+        """
+            Adds many products to the list if they are not exists.
+        """
+        for product in args:
+            if product in self.__products:
+                raise ValueError(f"Product {product} is already in the list")
```

Comment:
let's avoid Errors while adding. Implement it in such a way: if it is already on the list nothing happens, else add a new one. 

Tip:
Python already have special datatype for this))